src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1

```diff
@@ -862,71 +862,52 @@ class Project: ADOSVTBase
-                        $permissionsContextJson = $response.dataProviders.'ms.vss-admin-web.security-view-data-provider'.permissionsContextJson
-                        $permissionsContextJson = $permissionsContextJson | ConvertFrom-Json
-                        if ($permissionsContextJson.inheritPermissions) {
-                            $failedRepos += $repo.name
-                        }
-                        else {
-                            $passedRepos += $repo.name
-                        }
+        #permissionSetId = '2e9eb7ed-3c0a-47d4-87c1-0ffdd275fd87' is the std. namespaceID. Refer: https://docs.microsoft.com/en-us/azure/devops/organizations/security/manage-tokens-namespaces?view=azure-devops#namespaces-and-their-ids
+        $repoNamespaceId = '2e9eb7ed-3c0a-47d4-87c1-0ffdd275fd87'
+        try
+        {
+            $repoPermissionUrl = 'https://dev.azure.com/{0}/_apis/accesscontrollists/{1}?api-version=6.0' -f $this.OrganizationContext.OrganizationName, $repoNamespaceId;
+            $responseObj = [WebRequestHelper]::InvokeGetWebRequest($repoPermissionUrl)
+            if (-not [string]::IsNullOrEmpty($responseObj) -and ($responseObj | Measure-Object).Count -gt 0)
+            {
+                $repoDefnsObj = $this.FetchRepositoriesList()
+                $failedRepos = @()
+                $passedRepos = @()
+                foreach ($repo in $repoDefnsObj)
+                {
+                    $repoToken = "repoV2" + "/" + $projectId + "/" + $repo.id
+                    $repoObj = $responseObj | where-object {$_.token -eq $repoToken}
+                    if ($null -ne $repoObj -and ($repoObj | Measure-Object).Count -gt 0 -and $repoObj.inheritPermissions)
+                    {
+                        $failedRepos += $repo.name
                     }
-                    else {
-                        $controlResult.AddMessage([VerificationResult]::Error, "Could not fetch details for repository $($repo.name). Please verify from portal whether inherited permissions is disabled.");
+                    else
+                    {
+                        $passedRepos += $repo.name
                     }
                 }
-                catch {
-                    $controlResult.AddMessage([VerificationResult]::Error, "Could not fetch details for repository $($repo.name). Please verify from portal whether inherited permissions is disabled. $($_).");
+
+                $failedcount = $failedRepos.Count
+                $passedcount = $passedRepos.Count
+                $passedRepos = $passedRepos | sort-object
+                if($failedcount -gt 0)
+                {
+                    $failedRepos = $failedRepos | sort-object
+                    $controlResult.AddMessage([VerificationResult]::Failed, "Inherited permissions are enabled on the below $($failedcount) repositories:", $failedRepos);
```

Comment:
What if count is 1? The message will be printed as  below which is gramatically wrong
"Inherited permissions are enabled on the below 1 repositories"
Instead can we add separate message to control result for count?

Reply:
changes done